core/src/main/java/com/novoda/merlin/receiver/ConnectivityCallbacks.java

```diff
@@ -35,15 +35,23 @@ public void onLost(Network network) {
     }
 
     private void notifyMerlinService(Network network) {
+        if (connectivityChangesNotifier.canNotify()) {
```

Comment:
Both `ConnectivityCallbacks` and `ConnectivityReceiverConnectivityChangeNotifier` check that they are able to `notify`, where `notify` is where the `Service` is bound. If the `Service` is bound then it's `Forwarder` must be present.